browser/net/global_privacy_control_network_delegate_helper.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/net/global_privacy_control_network_delegate_helper.h"
+
+#include <memory>
+
+#include "base/feature_list.h"
+#include "brave/common/brave_features.h"
+#include "brave/common/network_constants.h"
+#include "net/base/net_errors.h"
+#include "net/http/http_request_headers.h"
+
+namespace brave {
+
+int OnBeforeStartTransaction_GlobalPrivacyControlWork(
+    net::HttpRequestHeaders* headers,
+    const ResponseCallback& next_callback,
+    std::shared_ptr<BraveRequestInfo> ctx) {
+  if (base::FeatureList::IsEnabled(features::kGlobalPrivacyControl)) {
+    headers->SetHeader(kSecGpcHeader, "1");
```

Comment:
do we need it for all requests, or just for navigations/downloads?

Reply:
Talked to @pes10k about this as it's not currently described anywhere - he will try to get that clarified in the spec but there shouldn't be any harm in doing it this way.

Reply:
just heard back, its intended to be on every request